gcloud/pubsub/topic.py

```diff
@@ -278,6 +278,31 @@ def get_iam_policy(self, client=None):
         resp = client.connection.api_request(method='GET', path=path)
         return Policy.from_api_repr(resp)
 
+    def set_iam_policy(self, policy, client=None):
+        """Update the IAM policy for the topic.
```

Comment:
They feel the same to me:  we use `update` to mean "replace completely" vs. `patch` (replace only changed bits):  this is a complete replacement.

The [expectation in the docs](https://cloud.google.com/pubsub/reference/rest/Shared.Types/Policy#FIELDS.etag) is that one will first fetch the policy, update it in place, and then send the update, so that one preserves the `ETag`.